sources/Algo/include/Algo.h

```diff
@@ -952,5 +953,76 @@ class StaticVarCompensatorAlgorithm : public NodeAlgorithm {
   StaticVarCompensatorDefinitions& svarcsDefinitions_;  ///< the static var compensator definitions to update
 };
 
+/**
+ * @brief Contingencies valid for simulation
+ */
+class ValidContingencies {
+ public:
+  using ContingencyId = std::string;                                         ///< Alias for contingency identifier
+  using ElementId = std::string;                                             ///< Alias for element identifier in contingency
+  using ContingencyRef = std::reference_wrapper<const inputs::Contingency>;  ///< Alias for reference to contingency
+
+  /**
+   * @brief Constructor
+   * @param contingencies The list of contingencies given in the inputs
+   */
+  explicit ValidContingencies(std::shared_ptr<const std::vector<inputs::Contingency>> contingencies);
```

Comment:
Use reference instead of pointer when possible, because null pointers exists but not "null" references

Reply:
Changed to a reference